sstables/open_info.hh

```diff
@@ -34,6 +34,8 @@ struct entry_descriptor {
     sstable_format_types format;
     component_type component;
     std::optional<sstable_state> state;
+    // an optional path component after the prefix offered by storage_options
+    std::optional<sstring> prefix;
```

Comment:
The entry_descriptor is too generic place for special-purpose "prefix" thing. Is it possible to re-use prefix from restore_components_lister?